common/cache/domainCache.go

```diff
@@ -411,15 +420,12 @@ func (c *DefaultDomainCache) refreshLoop() {
 		case <-c.shutdownChan:
 			return
 		case <-timer.Chan():
-			for err := c.refreshDomains(); err != nil; err = c.refreshDomains() {
-				select {
-				case <-c.shutdownChan:
-					return
-				default:
-					c.logger.Error("Error refreshing domain cache", tag.Error(err))
-					c.timeSource.Sleep(DomainCacheRefreshFailureRetryInterval)
-				}
+			err := c.refreshDomains()
+			if err != nil {
+				c.logger.Error("Error refreshing domain cache", tag.Error(err))
```

Comment:
We could consider logging fatal here? Since we already retryed quite a lot, and we still failed.

Reply:
Let's leave it like this. It may have failed due to persistence rps limits and in those cases we don't want to get into crash loop. Ideally fetching domains shouldn't be subject to the same limits. @davidporter-id-au will look into granular rate limits area.